homeassistant/components/apple_tv/__init__.py

```diff
@@ -370,19 +314,77 @@ def _async_setup_device_registry(self):
         )
 
     @property
-    def is_connecting(self):
+    def is_connecting(self) -> bool:
         """Return true if connection is in progress."""
         return self._task is not None
 
-    def _address_updated(self, address):
+    def _address_updated(self, address: str) -> None:
         """Update cached address in config entry."""
         _LOGGER.debug("Changing address to %s", address)
         self.hass.config_entries.async_update_entry(
             self.config_entry, data={**self.config_entry.data, CONF_ADDRESS: address}
         )
 
-    def _dispatch_send(self, signal, *args):
+    def _dispatch_send(self, signal: str, *args: Any) -> None:
         """Dispatch a signal to all entities managed by this manager."""
         async_dispatcher_send(
             self.hass, f"{signal}_{self.config_entry.unique_id}", *args
         )
+
+
+class AppleTVEntity(Entity):
```

Comment:
Here is the diff:

```diff
diff --git a/homeassistant/components/apple_tv/apple_tv_entity.py b/homeassistant/components/apple_tv/apple_tv_entity.py
index cabe9dd276..9c53ee1a4f 100644
--- a/homeassistant/components/apple_tv/apple_tv_entity.py
+++ b/homeassistant/components/apple_tv/apple_tv_entity.py
@@ -4,10 +4,10 @@ class AppleTVEntity(Entity):
     _attr_should_poll = False
     _attr_has_entity_name = True
     _attr_name = None
+    atv: AppleTV | None = None
 
-    def __init__(self, name, identifier, manager):
+    def __init__(self, name: str, identifier: str, manager: AppleTVManager) -> None:
         """Initialize device."""
-        self.atv = None
         self.manager = manager
         self._attr_unique_id = identifier
         self._attr_device_info = DeviceInfo(
@@ -15,18 +15,18 @@ class AppleTVEntity(Entity):
             name=name,
         )
 
-    async def async_added_to_hass(self):
+    async def async_added_to_hass(self) -> None:
         """Handle when an entity is about to be added to Home Assistant."""
 
         @callback
-        def _async_connected(atv):
+        def _async_connected(atv: AppleTV) -> None:
             """Handle that a connection was made to a device."""
             self.atv = atv
             self.async_device_connected(atv)
             self.async_write_ha_state()
 
         @callback
-        def _async_disconnected():
+        def _async_disconnected() -> None:
             """Handle that a connection to a device was lost."""
             self.async_device_disconnected()
             self.atv = None
@@ -49,8 +49,8 @@ class AppleTVEntity(Entity):
             )
         )
 
-    def async_device_connected(self, atv):
+    def async_device_connected(self, atv: AppleTV) -> None:
         """Handle when connection is made to device."""
 
-    def async_device_disconnected(self):
+    def async_device_disconnected(self) -> None:
         """Handle when connection was lost to device."""

```

Reply:
I believe the reason you had to move the class is because you forgot to add `from __future__ import annotations`. I moved it back when fixing the merge conflicts just now.